pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -377,7 +379,6 @@ boolean markAckForBatchMessage(BatchMessageIdImpl batchMessageId, AckType ackTyp
                 ackMessagesInEarlierBatch(batchMessageId, message);
             }
             if (log.isDebugEnabled()) {
-                int outstandingAcks = batchMessageAckTracker.get(message).cardinality();
```

Comment:
Since this is only needed for debug, could we just re-acquire the lock here? (or don't even care about it). 
I think `bitSet.cardinality()` might be expensive in that it's kind of `O(n)`